controllers/storageclassrequest/storageclassrequest_controller.go

```diff
@@ -189,6 +189,23 @@ func (r *StorageClassRequestReconciler) reconcilePhases() (reconcile.Result, err
 		return reconcile.Result{}, err
 	}
 
+	profileName := r.StorageClassRequest.Spec.StorageProfile
+	if profileName == "" {
+		profileName = r.storageCluster.Spec.DefaultStorageProfile
+	}
+
+	// Fetch StorageProfile by name in the StorageCluster's namespace
+	storageProfile := v1.StorageProfile{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      profileName,
+			Namespace: r.storageCluster.Namespace,
+		},
+	}
+
+	if err := r.get(&storageProfile); err != nil {
+		return reconcile.Result{}, fmt.Errorf("no storage profile CR found for storage profile %s", profileName)
```

Comment:
return or log `err` as well and `get` failure can be due to other errors as well, its not always `Not Found` error isn't it?